app/models/ems_refresh/save_inventory_container.rb

```diff
@@ -186,16 +187,52 @@ def save_container_env_vars_inventory(container_definition, hashes, target = nil
               else
                 []
               end
-
     save_inventory_multi(:container_env_vars, container_definition, hashes, deletes, [:name, :value, :field_path])
     store_ids_for_new_records(container_definition.container_env_vars, hashes, [:name, :value, :field_path])
   end
 
+  def save_container_images_inventory(ems, hashes, target = nil)
+    return if hashes.nil?
+
+    ems.container_images(true)
+    deletes = if target.kind_of?(ExtManagementSystem)
+                ems.container_images.dup
+              else
+                []
+              end
+
+    hashes.each do |h|
+      h[:container_image_registry_id] = h[:container_image_registry][:id] unless h[:container_image_registry].nil?
+    end
+
+    save_inventory_multi(:container_images, ems, hashes, deletes, [:image_ref, :container_image_registry_id], [],
+                         :container_image_registry)
+    store_ids_for_new_records(ems.container_images, hashes,
+                              [:image_ref, :container_image_registry_id])
+  end
+
+  def save_container_image_registries_inventory(ems, hashes, target = nil)
+    return if hashes.nil?
+    target = ems if target.nil?
+
+    ems.container_image_registries(true)
+    deletes = if target.kind_of?(ExtManagementSystem)
+                ems.container_image_registries.dup
+
```

Comment:
@zeari Please remove this blank line to match other code in this file with the same pattern.